org-roam.el

```diff
@@ -587,6 +590,47 @@ This needs to be quick/infrequent, because this is run at
     (call-process org-roam-graphviz-executable nil 0 nil temp-dot "-Tsvg" "-o" temp-graph)
     (call-process org-roam-graph-viewer nil 0 nil temp-graph)))
 
+(defun org-roam--rename-file-links (file new-file &rest args)
+  "Rename backlinks of FILE to refer to NEW-FILE."
+  (when (and (not (auto-save-file-name-p file))
+             (not (auto-save-file-name-p new-file))
+             (org-roam--org-roam-file-p new-file))
+    (org-roam--ensure-cache-built)
+    (org-roam--clear-file-from-cache file)
+
+    (let* ((files (gethash file org-roam-backward-links-cache nil))
+           (path (file-truename file))
+           (new-path (file-truename new-file))
+           (slug (org-roam--get-title-or-slug file))
+           (old-title (format org-roam-link-title-format slug))
+           (new-slug (or (org-roam--get-title-from-cache path)
+                         (org-roam--get-title-or-slug new-path)))
+           (new-title (format org-roam-link-title-format new-slug)))
+      (when files
+        (save-excursion
+          (maphash (lambda (file-from contents)
+                     (let* ((file-dir (file-name-directory file-from))
+                            (relative-path (file-relative-name new-path file-dir))
+                            (old-relative-path (file-relative-name path file-dir))
+                            (slug-regex (regexp-quote (format "[[file:%s][%s]]" old-relative-path old-title)))
+                            (named-regex (concat
+                                          (regexp-quote (format "[[file:%s][" old-relative-path))
+                                          "\\(.*\\)"
+                                          (regexp-quote "]]"))))
+                       (with-temp-file file-from
+                         (insert-file-contents file-from)
+                         (while (re-search-forward slug-regex  nil t)
+                           (replace-match (format "[[file:%s][%s]]" relative-path new-title)))
+                         (goto-char (point-min))
+                         (while (re-search-forward named-regex nil t)
+                           (replace-match (format "[[file:%s][\\1]]" relative-path))))
+                       (find-file file-from)
+                       (org-roam--update-cache)))
+                   files))))))
+
+(advice-add 'rename-file :after 'org-roam--rename-file-links)
```

Comment:
It's a bit inappropriate for the advice to out here like this, I think eventually `org-roam-mode` needs to be made a global mode and the advice to be added and removed via toggling the global mode. But I'm good with merging this as is first, and then fixing it later.